src/state/restock/restockState.ts

```diff
@@ -0,0 +1,6 @@
+import {StateBase} from '../base';
+import {RestockData} from './restockData';
+
+export type RestockState = StateBase & {
+    restockData: RestockData | [];
```

Comment:
No, not like this. More like 
```typescript
export type RestockState = StateBase & {
    restockData: RestockData;
}
```
and
```typescript
const initialState: RestockState = {
    restockData: [];
}
```

----

Basically you match your data with the data type, but not allowing that "fixed" data type to be one of the allowed.